### PR TITLE
Add monitoring DAG only in non-hosted orgs

### PR DESCRIPTION
**Pull request by feluelle**
## Description

> Describe the purpose of this pull request.

We want to remove the monitoring DAG for hosted clusters, hence we are only adding the monitoring DAG if the org is not a hosted org, the deployment type is not hosted dedicated, and not hosted shared.

## 🎟 Issue(s)

Related https://github.com/astronomer/laminar/issues/360

Depends on https://github.com/astronomer/astro/pull/14203

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [ ] Rebased from the main (or release if patching) branch (before testing)
- [ ] Ran `make test` before taking out of draft
- [ ] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: cloud/deploy/deploy.go [hunk 1]
```diff
@@ -140,14 +141,16 @@ func deployDags(path, dagsPath, runtimeID string, client astro.Client) (string,
 	// Check the dags directory
 	monitoringDagPath := filepath.Join(dagsPath, "astronomer_monitoring_dag.py")
 
-	// Create monitoring dag file
-	err := fileutil.WriteStringToFile(monitoringDagPath, airflow.MonitoringDag)
-	if err != nil {
-		return "", err
+	if !organization.IsOrgHosted() {
```
Comment (ianbuss) [inline cloud/deploy/deploy.go:144]: Does this cover both `HostedDedicated` and `HostedShared` deployments?

**Review by ianbuss — COMMENTED** [review]
One q

File: cloud/deploy/deploy.go (hunk 1 above)
Comment (feluelle) [reply cloud/deploy/deploy.go:144]: I think so - as long as they are in a hosted org.

Comment (kushalmalani) [reply cloud/deploy/deploy.go:144]: can we also add check for deploymentType
if !organization.IsOrgHosted() && !(deployment.IsDeploymentDedicated(deploymentType) || deployment.IsDeploymentHosted(deploymentType) {

Comment (feluelle) [reply cloud/deploy/deploy.go]: Done in 6d49e15

**Review by ianbuss — APPROVED** [review]
One nit around the conditional function determining if the monitoring dag is sent. Since the ultimate aim is to remove the monitoring DAG even for Hybrid it might be nice to have this logic in one function to make it easy to change. Otherwise it looks great.

File: cloud/deploy/deploy.go [hunk 2]
```diff
@@ -136,18 +138,20 @@ func removeDagsFromDockerIgnore(fullpath string) error {
...
-func deployDags(path, dagsPath, runtimeID string, client astro.Client) (string, error) {
+func deployDags(path, dagsPath, deploymentType, runtimeID string, client astro.Client) (string, error) {
 	// Check the dags directory
 	monitoringDagPath := filepath.Join(dagsPath, "astronomer_monitoring_dag.py")
 
-	// Create monitoring dag file
-	err := fileutil.WriteStringToFile(monitoringDagPath, airflow.MonitoringDag)
-	if err != nil {
-		return "", err
+	if !organization.IsOrgHosted() && !deployment.IsDeploymentDedicated(deploymentType) && !deployment.IsDeploymentHosted(deploymentType) {
```
[the remark this answers is not in the document]
Comment (feluelle) [reply cloud/deploy/deploy.go:145]: Good call!

[the remark this answers is not in the document]
Comment (feluelle) [reply cloud/deploy/deploy.go]: Done in 158dc4e

**Review by neel-astro — APPROVED** [review]

**Pull request merged**